plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/jmx/OpenshiftConnectionProvider.java

```diff
@@ -0,0 +1,147 @@
+package org.jboss.tools.openshift.core.jmx;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.management.remote.JMXConnectorProvider;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.debug.core.ILaunchManager;
+import org.eclipse.wst.server.core.IServer;
+import org.eclipse.wst.server.core.ServerEvent;
+import org.jboss.ide.eclipse.as.core.server.UnitedServerListener;
+import org.jboss.ide.eclipse.as.core.server.UnitedServerListenerManager;
+import org.jboss.tools.jmx.core.AbstractConnectionProvider;
+import org.jboss.tools.jmx.core.IConnectionWrapper;
+import org.jboss.tools.openshift.core.server.OpenShiftServer;
+import org.jboss.tools.openshift.internal.core.OpenShiftCoreActivator;
+
+public class OpenshiftConnectionProvider extends AbstractConnectionProvider {
+	private class ServerListener extends UnitedServerListener {
+		@Override
+		public boolean canHandleServer(IServer server) {
+			return true;
+		}
+		
+		public void serverChanged(ServerEvent event) {
+			updateConnection(event.getServer());
+		}
+
+		public void serverAdded(IServer server) {
+			updateConnection(server);
+		}
+
+		public void serverChanged(IServer server) {
+			updateConnection(server);
+		}
+
+		public void serverRemoved(IServer server) {
+			updateConnection(server);
+		}
+	}
+
+
+	public static final String ID = "com.jboss.tools.jmx.core.remoting";
+
+	private Map<String, IConnectionWrapper> idToConnection = new HashMap<String, IConnectionWrapper>();
+
+	public OpenshiftConnectionProvider() {
+		UnitedServerListenerManager.getDefault().addListener(new ServerListener());
+	}
+
+	@Override
+	public String getId() {
+		return ID;
+	}
+
+	@Override
+	public String getName(IConnectionWrapper wrapper) {
+		if (wrapper instanceof OpenshiftConnectionWrapper) {
+			return ((OpenshiftConnectionWrapper) wrapper).getName();
+		}
+		return null;
+	}
+
+	private IConnectionWrapper createConnection(IServer server) throws MalformedURLException {
+		ConnectorProviderRegistry connectorProviders = OpenShiftCoreActivator.getDefault().getConnectorProviders();
+		VersionKey version = new VersionDetector(server).guess();
+		if (version != null) {
+			JMXConnectorProvider provider = connectorProviders.getProvider(version);
+			return new OpenshiftConnectionWrapper(this, server, provider);
+		}
+		return null;
+	}
+
+	@Override
+	public IConnectionWrapper[] getConnections() {
+		return idToConnection.values().toArray(new IConnectionWrapper[idToConnection.values().size()]);
+	}
+
+	@Override
+	public boolean canCreate() {
+		return false;
+	}
+
+	@Override
+	public boolean canDelete(IConnectionWrapper wrapper) {
+		return false;
+	}
+
+	@Override
+	public IConnectionWrapper createConnection(@SuppressWarnings("rawtypes") Map map) throws CoreException {
+		throw new UnsupportedOperationException("Cannot create connection");

```

Comment:
cant implement this (what about the #createConnection that we already have, which is close)? no added value?